### PR TITLE
Flask required only if necessary

### PR DESCRIPTION
**Pull request by rafaelcaricio**
Related to #380

Only requires Flask package to be installed when using specifically related functionality.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -63,7 +64,7 @@ Why Connexion
 
 With Connexion, you write the spec first. Connexion then calls your Python
 code, handling the mapping from the specification to the code. This
-incentivizes you to write the specification so that all of your
+incentives you to write the specification so that all of your
```
Comment (hjacobs) [inline README.rst]: I think "incentivizes" was correct!

Comment (rafaelcaricio) [inline README.rst]: Changing it back...

**Comment by hjacobs** [comment]
@rafaelcaricio can we first release Connexion without this breaking change? This breaking change (existing users need to update their requirements to pull in Flask) should also get a major version number bump I guess. Let's first roll out the Framework changes in a compatible way (no changes necessary by users) and see if it works for everybody.

**Comment by rafaelcaricio** [comment]
@hjacobs I was thinking about that. But still this changes here are important. I will just change the `setup.py` for now to include Flask by default and change back the README.

**Comment by rafaelcaricio** [comment]
@hjacobs Done in https://github.com/zalando/connexion/pull/424/commits/86891ab74df0bdbf68051327432e7fe05a026cd2.

**Comment by rafaelcaricio** [comment]
Changes in this PR will make easier to release Connexion 2.x in near future. I still want to make some more changes before going for 2.x, mostly related to https://github.com/zalando/connexion/issues/283. So we release together those breaking changes.

**Comment by hjacobs** [comment]
@rafaelcaricio we could release the current master as version 1.*. What do you think?

**Comment by vimalloc** [comment]
(totally selfish here), but if at all possible, I would like to get #425 in the next released version. Currently, we are doing a pip install from the forked version, but it isn't ideal.

**Comment by hjacobs** [comment]
@rafaelcaricio after the latest merge -> conflicts to be resolved :smirk:

**Comment by vimalloc** [comment]
Sorry about that :( Looks like it is just imports, shouldn't be too hard to fix. I *think* after the resolution it should look like

```python
￼from flask import jsonify, redirect, request
￼  
￼from connexion import NoContent, ProblemException, problem
```

**Comment by rafaelcaricio** [comment]
@vimalloc Don't worry. I am fixing that. 😉 

@hjacobs Yes, after merging this PR we can release a new version of Connexion still under 1.x. If people are not playing with Connexion's internals in their services everything should continue work as usual.

**Comment by hjacobs** [comment]
@rafaelcaricio I felt adventurous and released https://github.com/zalando/connexion/releases/tag/1.1.6

**Comment by rafaelcaricio** [comment]
@hjacobs it is ok, but flask was removed from the `setup.py` in the current master. This PR puts it back. So after merging this, we should release again with this "bug fix".

**Comment by hjacobs** [comment]
@rafaelcaricio uh, I did not see this!

**Comment by hjacobs** [comment]
:+1:

**Comment by hjacobs** [comment]
@rafaelcaricio go for it, my GnuPG setup is also broken atm, so you can do `tag -s`.

**Pull request merged**